source/CustomNode.cpp

```diff
@@ -0,0 +1,57 @@
+#include "CustomNode.h"
+
+using namespace cugl;
+
+/** Position to hide stuff offscreen. */
+constexpr float OFF_SCREEN_POS = 1500;
+
+bool CustomNode::init(std::shared_ptr<DonutModel> player, float shipSize, float angle,
+					  float radius) {
+	playerDonutModel = player;
+	this->shipSize = shipSize;
+	this->angle = angle;
+	this->radius = radius;
+	Node::init();
+	return false;
+}
+
+void CustomNode::dispose() {
+	playerDonutModel = nullptr;
+	Node::dispose();
+}
+
+void CustomNode::draw(const shared_ptr<cugl::SpriteBatch>& batch, const cugl::Mat4& transform,
+					  cugl::Color4 tint) {
+	if (isActive()) {
+		// Model is currently active
+
+		prePosition();
+
+		float onScreenAngle = getOnScreenAngle(angle);
+		if (isComingIntoView(onScreenAngle)) {
+			// Entering visible range
+			float relativeAngle = onScreenAngle - getParent()->getParent()->getAngle();
```

Comment:
I'm a bit iffy about including this line in `CustomNode` since it's making assumptions about the scene graph structure (that `nearSpace` is two levels higher). But since we're pretty much finished with game mechanics, I guess we can leave it like this, but maybe make a note about it in the spec.

Reply:
I mean, the whole point of the `CustomNode` is that it is our custom node that fits into our scene graph. It makes sense for it to be adapted to our graph. That said, I did add more explicit documentation in `CustomNode.h` about this.